### ISSUE TITLE
Plan Quests

### ISSUE THREAD

**Issue by drusepth** [indentlabs/notebook#145]
Requested by a few of the more DnD-focused users. 

>It would be great to be able to "create" a new quest, including links to who the questgiver is, if there is an item as a reward, locations they need to go to, etc.

It'd be nice if we could figure out a way to make this generic enough to work with campaigns _and_ writing (like rolling it into Plots or something). Just making this issue so we can think about implementation.

**Comment by Cantido** [comment]
We might be able to generify it enough to where creating some other story content could also be used for DnD. For example, Characters are generic enough, and with a DnD "plugin" that can store a few integers, it's suddenly a character sheet.

**Comment by jsonnull** [comment]
To me, quests have a narrative chronology to them not unlike History sections of various components. It could be neat to exposes quests as well as history as timelines for editing.

**Comment by Cantido** [comment]
True, I think any story/history/plot thing could be a quest. Quests may have branching points that result in totally different canons, which would require some kind of mechanics that any other story-planning-based feature would not. Or, maybe it would and it would be super helpful for writers to see the branching paths and pick their favorite.

**Comment by jsonnull** [comment]
@Cantido instead of generalizing to "quests" and using that to expand on history functionality, expose a `timeline` where you can add events and reorder them—basically just a list. The timeline feature could be used in quests to begin with but also made available in character history, location history, world history, and possibly even reference other timelines from a timeline event. E.G. in the history of this world, there was a "campaign" event which directly refers to a quest.

I'm just tossing ideas into the hat here. Basically I see an opportunity to flesh out history editing as well as quests in the same way.

**Comment by jsonnull** [comment]
I think tree could overly complicate things. As a general rule, I like organizational strategies like this to be as distant from the database as possible—ideally, users have the tools available to intuitively create the structure they need within the app. This relieves the developer from the burden of maintaining a complex organizational scheme on the backend and keeps options open for different strategies later.

Applying this principle here I think the best solution would be if users created branches in quests as separate quests. For instance, if a group of adventurers go into town and from that point on they could set in motion one of several possible chains of events... create a separate quest for each possible branch and at the end of the trunk have an "event" in the timeline that says "Depending on what happens, go to [branch one](#) or [branch two](#)."

Maybe we can have an explicit "branch" or "subplot" timeline entry type that explicitly references another quest here. This is a great way to pull in these narratives in location or character history timelines too!

As an example, here's what a timeline might look like...

> [Episode One - The Adventure Begins](#)
> - Adventurers escape the prison
> - On the road out of town, adventurers confront bandits
> - When they arrive at the city, the guards are denying entry
> - [REF] If the adventurers find the secret way in, start [underground society plot](#)
> - [REF] If the adventurers bribe the guards, mention hooks for [corruption sub plot](#)

One final thought for now—by having an explicit reference entry, there's groundwork to eventually layer in a tree view or some other kind of interface for viewing child timelines.

What do you all think? Trying to keep this as simple as possible. Add a Quest category, introduce a timeline editing scheme with sortable events, extend history sections to use timeline editing, possibly allow referencing quests and timelines from other timelines.

**Comment by Cantido** [comment]
I think separate quests linked like that would be great. We have the Campaign idea which sounds like it ought to be the container for a bunch of quests like this.

**Comment by jsonnull** [comment]
That sounds excellent!

**Comment by drusepth** [comment]
Depends on #331 before we can start this.
